zookeeper/src/main/java/com/linecorp/armeria/server/zookeeper/ZooKeeperUpdatingListenerBuilder.java

```diff
@@ -0,0 +1,165 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.zookeeper;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static java.util.Objects.requireNonNull;
+
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.CuratorFrameworkFactory;
+
+import com.linecorp.armeria.client.Endpoint;
+import com.linecorp.armeria.common.zookeeper.NodeValueCodec;
+import com.linecorp.armeria.internal.zookeeper.Constants;
+import com.linecorp.armeria.server.ServerListener;
+
+/**
+ * Builds a new {@link ServerListener}, which registers the server to the zookeeper instance.
+ * <h2>Example</h2>
+ * <pre>{@code
+ * ServerListener register =
+ * // Zookeeper Connection String and Zookeeper Node to use
+ * new ZooKeeperUpdatingListenerBuilder("myZooKeeperHost:2181", "/myProductionEndpoints")
+ * // Session Timeout
+ * .sessionTimeOut(10000)
+ * // NodeValueCodec
+ * .nodeValueCodec(NodeValueCodec.DEFAULT)
+ * .build();
+ * // Set to `Server`.
+ * Server server = ...
+ * server.serverListener(register);
+ * }</pre>
+ *
+ * <p>You can also specify the {@link CuratorFramework} instance to use. In this case,
+ * {@link #connectTimeout(int)} and {@link #sessionTimeout(int)} will be ignored.
+ *
+ * <h2>Example</h2>
+ * <pre>{@code
+ * // CuratorFramework instance and Zookeeper Node to use
+ * ServerListener register = new ZooKeeperUpdatingListenerBuilder(curatorFramework, "/myProductionEndpoints")
+ * // NodeValueCodec
+ * .nodeValueCodec(NodeValueCodec.DEFAULT)
+ * .build();
+ * // Set to `Server`.
+ * Server server = ...
+ * server.serverListener(register);
+ * }</pre>
+ * */
+public final class ZooKeeperUpdatingListenerBuilder {
+    private CuratorFramework client;
+    private String connect;
+    private String node;
+    private int connectTimeout = Constants.DEFAULT_CONNECT_TIMEOUT;
+    private int sessionTimeout = Constants.DEFAULT_SESSION_TIMEOUT;
+    private Endpoint endpoint;
+    private NodeValueCodec nodeValueCodec = NodeValueCodec.DEFAULT;
+
+    /**
+     * Creates a {@link ZooKeeperUpdatingListenerBuilder} with a {@link CuratorFramework} instance and a zNode.
+     *
+     * @param client the curator framework instance.
+     */
+    public ZooKeeperUpdatingListenerBuilder(CuratorFramework client, String node) {
+        this.client = requireNonNull(client, "client");
+        checkArgument(!isNullOrEmpty(node), "node");
+        this.node = node;
+    }
+
+    /**
+     * Creates a {@link ZooKeeperUpdatingListenerBuilder} with a Zookeeper Connection String and a zNode.
+     *
+     * @param connect the zookeeper connection string.
+     * @param node the zookeeper node to register.
```

Comment:
- global comment: zookeeper, Zookeeper -> ZooKeeper
- global comment: Please remove the trailing dot (.) if the description is not a string and there's no follow-up sentence.